File: 884 - Uncommon Words from Two Sentences.js
/**
 * @param {string} A
 * @param {string} B
 * @return {string[]}
 */
var uncommonFromSentences = function (A, B) {
    let count = new Map();
    for (let word of A.split(' ')) {
        if (count.has(word)) {
            count.set(word, count.get(word) + 1);
        } else {
            count.set(word, 1);
        }
    }
    for (let word of B.split(' ')) {
        if (count.has(word)) {
            count.set(word, count.get(word) + 1);
        } else {
            count.set(word, 1);
        }
    }
    let res = [];
    for (let word of count.keys()) {
        if (count.get(word) == 1) {
            res.push(word);
        }
    }
    return res;
};

let b = "this apple is sweet";
let a = "this apple is sour";
let c = uncommonFromSentences(a, b);
console.log(c);

File: 884 - Uncommon Words from Two Sentences.cpp
#include <vector>
#include <string>
#include <unordered_map>


using namespace std;
class Solution {
public:
    vector<string> uncommonFromSentences(string A, string B) {
        A = A + ' ' + B + ' ';
        unordered_map<string, int> count;
        string word;
        for (char c : A) {
            if (c == ' ' && word.size() != 0) {
                count[word] += 1;
                word = "";
            } else {
                word += c;
            }
        }
        vector<string> res;
        for (auto i = count.begin(); i != count.end(); i++) {
            if (i->second == 1) {
                res.push_back(i->first);
            }
        }
        return res;
    }
};